rcl/src/rcl/wait.c

```diff
@@ -79,6 +80,9 @@ __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
     assert(ret == RCL_RET_OK);  // Defensive, shouldn't fail with size 0.
   }
   if (wait_set->impl) {
+    if (wait_set->impl->rmw_waitset) {
+      assert(rmw_destroy_waitset(wait_set->impl->rmw_waitset) == RMW_RET_OK);
```

Comment:
I'm following the pattern of the above code here. Since this function signature has no way of returning an error code, I'm not sure how to propagate the error to the highest level. @wjwwood any thoughts on how these functions should report errors in release builds?